content/glossary/_index.md

```diff
@@ -2,311 +2,411 @@
+## Leader
 
-### Message
+A leader, in the context of Filecoin consensus, is a node that is chosen to propose the next block in the blockchain during [Leader Electioni](#leader-election).
 
-A message is a call to an actor in the Filecoin VM.
+## Leader election
 
-### Miner
+Same as [Election](glossary#election).
+
+## Message
+
+A message is a call to an actor in the Filecoin VM. The term _message_ is used to refer to data stored as part of a [block](glossary#block). A block can contain several messages.
+
+## Miner
 
 A miner is an actor in the Filecoin system performing a service in the network for a reward.
 
-There are multiple types of miners in Filecoin:
+There are three types of miners in Filecoin:
 
-- Storage miners - storage miners
-- Retrieval miners:
-- Repair miners (to be split out):
+- [Storage miners](glossary#storage-miner-actor), who store files on behalf of clients.
+- [Retrieval miners](glossary#retrieval-miner), who deliver stored files to clients. 
+- Repair miners, who replicate files to keep them available in the network, when a storage miner presents a fault.
 
-### Node
+## Multisig Actor
 
-A node is a communication endpoint that implements the Filecoin protocol. (also mention IPLD Node?)
+The _Multisig Actor_ (or Multi-Signature Wallet Actor) is responsible for dealing with operations involving the Filecoin wallet.
 
-### On-chain/off-chain
+## Node
 
-### Online/offline
+A node is a communication endpoint that implements the Filecoin protocol.
 
-### Partial Ticket
+## On-chain/off-chain
 
-Ticket produced as part of the ElectionPoSt process used to both prove leader elections and prove storage of a given data replica through PoSt. At least one is included in every block header.
+On-chain actions are those that change the state of the tree and the blockchain and interact with the Filecoin VM. Off-chain actions are those that do not interact with the Filecoin VM.
 
-### Payment Channel
+## Payment Channel
 
-A payment channel is set up between actors in the Filecoin system to enable off-chain payments with on-chain guarantees, making settlement more efficient.
+A payment channel is set up between actors in the Filecoin system to enable off-chain payments with on-chain guarantees, making settlement more efficient. Payment channels are managed by the Payment Channel Actor, who is responsible for setting up and settling funds related to payment channels.
 
-### Piece
+## Piece
 
-A piece is a portion of a file that gets fitted into a sector.
+The main unit of account and negotiation for the data that a user wants to store on filecoin. A piece is a [CAR file](https://github.com/ipld/specs/blob/master/block-layer/content-addressable-archives.md#summary) produced by an IPLD DAG with its own *payload CID* and *piece CID*. A piece is not a unit of storage, that is, it can be of any size up to the size of a [sector](glossary#sector). If a piece is larger than a sector, then it has to be split in two (or more) pieces.
```

Comment:
This is not exactly accurate.

Two things:
- a piece is not gauranteed to be a CAR file - Lotus software assumes this to be the case, but there is nothing preventing one from making a storage deal for non-CAR data as long as it matches the PieceCID.
- a piece cannot be "of any size" -- it must be a size that slots into a sector -- i.e. a power of 2 when bit-padded. (so without bit padding it must be a (power of 2) * 127/128)
- so technically, even Lotus's pieces are not simply CAR data -- they are CAR data + a bunch of zeros to make it the right size